src/automatic_differentiation.jl

```diff
@@ -578,8 +578,17 @@ function curl(f::F, v::Vec{3}) where F
     end
     return c
 end
-curl(f::F, v::Vec{1, T}) where {F, T} = curl(f, Vec{3}((v[1], T(0), T(0))))
-curl(f::F, v::Vec{2, T}) where {F, T} = curl(f, Vec{3}((v[1], v[2], T(0))))
+function curl(f::F, v::Vec{2, T}) where {F, T}
+    @inbounds begin
+        ∇f = gradient(f, v)
+        c = Vec{3}((zero(T), zero(T), ∇f[2,1] - ∇f[1,2]))
+    end
+    return c
+end
+function curl(f::F, v::Vec{1, T}) where {F, T}
+    TR = promote_type(T, eltype(f(v)))
+    return zero(Vec{3,  TR})
```

Comment:
```suggestion
    return zero(Vec{3, eltype(f(v))}) / oneunit(T)
```
(Gives correct type even if type includes units, however, gradient doesn't currently support units from Unitful.jl)